src/protocol.cpp

```diff
@@ -190,6 +190,29 @@ const std::vector<std::string> &getAllNetMessageTypes()
     return allNetMessageTypesVec;
 }
 
+int messageTypeToIndex(const std::string& msg_type)
+{
+    const std::vector<std::string>& message_types = getAllNetMessageTypes();
+    auto msg_type_itr = std::find(message_types.begin(), message_types.end(), msg_type);
+
+    if (msg_type_itr != message_types.end()) {
+        int index = msg_type_itr - message_types.begin();
+        return index;
+    } else {
+        // assign the last index to be a catch all for NET_MESSAGE_TYPE_OTHER
+        return NUM_NET_MESSAGE_TYPES;
+    }
+}
+
+const std::string& messageTypeFromIndex(size_t index)
```

Comment:
Good point. Do you think it's okay to do 

```
    if (index >= NUM_NET_MESSAGE_TYPES) {
        return NET_MESSAGE_TYPE_OTHER;
    }
```

Or should I add a separate case for `if (index > NUM_NET_MESSAGE_TYPES)` that throws an error? Not in love with this because it's harder to get test coverage, but it certainly feels like the more correct thing to do.